pkg/reconciler/pipelinerun/metrics_test.go

```diff
@@ -102,6 +101,8 @@ func TestRecordPipelineRunDurationCount(t *testing.T) {
 			assertErrIsNil(err, "DurationAndCount recording recording got an error", t)
 			metricstest.CheckDistributionData(t, "pipelinerun_duration_seconds", test.expectedTags, 1, test.expectedDuration, test.expectedDuration)
 			metricstest.CheckCountData(t, "pipelinerun_count", test.expectedTags, test.expectedCount)
+
+			unregisterMetrics()
```

Comment:
This means metrics won't be unregistered if any of the above t.Fatal or panic. Is that intended?

Reply:
When `defer`ed, `unregister` is called after each `t.Run`, I think that's the correct behavior isn't it?

Reply:
> This means metrics won't be unregistered if any of the above t.Fatal or panic. Is that intended?

@ImJasonH you are right, I didnt realised the other case. 

>When defered, unregister is called after each t.Run, I think that's the correct behavior isn't it?

That's how it should be however looks like its not working the intended way. Im not getting concrete proof about it but I guess that is what happening

Reply:
Ahh you're right @ImJasonH  its `metricstest.Unregister` could behave asynchronously, defer is working intended way.